widgets/common/src/UpdateDeploymentModal.jsx

```diff
@@ -36,12 +33,16 @@ function UpdateDeploymentModal({ deployment, open, onHide, toolbox }) {
                 .then(fetchedBlueprint => {
                     const newDeploymentInputs = {};
                     const currentDeploymentInputs = deployment.inputs;
+                    const { data_types: dataTypes, inputs: plannedDeploymentInputs } = fetchedBlueprint.plan;
 
-                    _.forEach(fetchedBlueprint.plan.inputs, (inputObj, inputName) => {
-                        newDeploymentInputs[inputName] = Stage.Common.InputsUtils.getInputFieldInitialValue(
+                    _.forEach(plannedDeploymentInputs, (inputObj, inputName) => {
+                        const { type } = inputObj;
+                        const dataType = !_.isEmpty(dataTypes) && type ? dataTypes[type] : undefined;
```

Comment:
Isn't this equivalent to

```suggestion
                        const dataType = dataTypes?.[type];
```